src/cli/src/http.rs

```diff
@@ -0,0 +1,35 @@
+use async_h1::client;
+use http_types::{Error, Request, Response, StatusCode};
+use tracing::debug;
+
+pub async fn execute(req: Request) -> Result<Response, Error> {
+    debug!("executing request: {:#?}", req);
+
+    let host = req
+        .url()
+        .host_str()
+        .ok_or_else(|| Error::from_str(StatusCode::BadRequest, "missing hostname"))?
+        .to_string();
+
+    let is_https = req.url().scheme() == "https";
+
+    let addr = req
+        .url()
+        .socket_addrs(|| if is_https { Some(443) } else { Some(80) })?
+        .into_iter()
+        .next()
+        .ok_or_else(|| Error::from_str(StatusCode::BadRequest, "missing valid address"))?;
+
+    let tcp_stream = fluvio_future::net::TcpStream::connect(addr).await?;
+
+    let result = if is_https {
+        let tls_connector = fluvio_future::tls::TlsConnector::default();
```

Comment:
observe feature flag: `rust_tls` and `native_tls`.  `tls` should be under `rust_tls`.  For `native_tls`, use `fluvio_future::native_tls::TlsConnector`

Reply:
I will start looking at this. This raises a question to me though, I know there's been a lot of work on fluvio_future lately, are there any notes on how to properly set this type of thing up? I don't really know what's changed so as I'm looking at this I'm needing to dig through source and changes and feature flags to figure it out.

Reply:
I ended up just creating a function to return a TlsConnector of either implementation based on feature flag. I feel like that is the type of concern which might be better to push down into fluvio-future itself. We can have a discussion on that later perhaps

Reply:
Sure.  Changes were minimized to make sure existing code were not impacted.   We can add further improvements to `fluvio-future` and make it easier to consume it